File: frontend/src/i18n/translations.js
// Bản dịch tiếng Anh. Khóa (key) chính là chuỗi tiếng Việt gốc:
// - lang === 'vi'  → trả về chính khóa
// - lang === 'en'  → tra cứu trong bảng dưới, không có thì fallback về khóa
// Placeholder dạng {name} được thay ở hàm t().
export const en = {
  // --- Điều hướng / Sidebar ---
  'Bảng điều khiển': 'Dashboard',
  'Nhân viên': 'Employees',
  'Dự án': 'Projects',
  'Công việc': 'Tasks',
  'Chấm công': 'Attendance',
  'Văn phòng': 'Offices',
  'AI Gợi ý': 'AI Suggestions',
  'Công việc của tôi': 'My Tasks',
  'Chấm công của tôi': 'My Attendance',
  'Người dùng': 'Users',
  'Đăng xuất': 'Log out',
  'Quản lý': 'Manager',
  'Quản trị viên': 'Administrator',

  // --- Đăng nhập ---
  'Đăng nhập thất bại. Vui lòng kiểm tra lại email và mật khẩu.':
    'Login failed. Please check your email and password.',
  'Chào mừng trở lại': 'Welcome back',
  'Đăng nhập vào Task Manager': 'Sign in to Task Manager',
  'Mật khẩu': 'Password',
  'Đang đăng nhập...': 'Signing in...',
  'Đăng nhập': 'Sign in',
  'Chưa có tài khoản?': "Don't have an account?",
  'Đăng ký ngay': 'Sign up now',

  // --- Đăng ký ---
  'Mật khẩu xác nhận không khớp.': 'Password confirmation does not match.',
  'Đăng ký thành công! Đang chuyển đến trang đăng nhập...':
    'Registration successful! Redirecting to the login page...',
  'Đăng ký thất bại. Vui lòng thử lại.': 'Registration failed. Please try again.',
  'Tên người dùng': 'Username',
  'Xác nhận mật khẩu': 'Confirm password',
  'Tạo tài khoản': 'Create account',
  'Tham gia Task Manager hôm nay': 'Join Task Manager today',
  'Đang đăng ký...': 'Signing up...',
  'Đăng ký': 'Sign up',
  'Đã có tài khoản?': 'Already have an account?',

  // --- Dashboard ---
  'Tổng quan hệ thống': 'System overview',
  '{tasks} công việc đã được tạo, {done} đã hoàn thành':
    '{tasks} tasks created, {done} completed',
  'Tỷ lệ hoàn thành': 'Completion rate',
  'Chấm công hôm nay': "Today's attendance",
  'Phân bổ trạng thái công việc': 'Task status distribution',
  'Chưa có công việc': 'No tasks yet',
  'Nhân viên theo phòng ban': 'Employees by department',
  'Chưa có dữ liệu nhân viên': 'No employee data yet',
  'Số nhân viên': 'Number of employees',
  'Khác': 'Other',
  'Xin chào, {name}!': 'Hello, {name}!',
  'Bạn có {tasks} công việc được giao, {done} đã hoàn thành.':
    'You have {tasks} assigned tasks, {done} completed.',
  'Chờ xử lý': 'Pending',
  'Đang thực hiện': 'In progress',
  'Đã hoàn thành': 'Completed',
  'Chấm công tháng này': "This month's attendance",
  'Công việc của tôi theo trạng thái': 'My tasks by status',
  'Bạn chưa có công việc nào': "You don't have any tasks",
  'Công việc sắp đến hạn': 'Upcoming deadlines',
  'Không có công việc nào sắp đến hạn': 'No upcoming deadlines',

  // --- Trạng thái dùng chung ---
  'Hoàn thành': 'Completed',
  'Hoạt động': 'Active',
  'Đúng tiến độ': 'On track',
  'Quá hạn': 'Overdue',

  // --- Hành động dùng chung ---
  'Hành động': 'Actions',
  'Sửa': 'Edit',
  'Xóa': 'Delete',
  'Hủy': 'Cancel',
  'Lưu': 'Save',
  'Đang lưu...': 'Saving...',
  'Xóa thất bại.': 'Delete failed.',
  'Lưu thất bại.': 'Save failed.',
  'Mô tả': 'Description',
  'Trạng thái': 'Status',
  'Lỗi:': 'Error:',

  // --- Nhân viên ---
  'Bạn có chắc muốn xóa nhân viên này?': 'Are you sure you want to delete this employee?',
  'Thêm nhân viên': 'Add employee',
  'Tìm kiếm theo tên, phòng ban...': 'Search by name or department...',
  'Họ': 'First name',
  'Tên': 'Last name',
  'Chức vụ': 'Position',
  'Phòng ban': 'Department',
  'Không có nhân viên nào.': 'No employees found.',
  'Chỉnh sửa nhân viên': 'Edit employee',
  'Thêm nhân viên mới': 'Add new employee',
  'VD: Kỹ sư phần mềm': 'e.g. Software Engineer',
  'VD: IT, Kế toán, Nhân sự': 'e.g. IT, Accounting, HR',
  'Kỹ năng': 'Skills',
  'VD: Java, Spring Boot, PostgreSQL, React': 'e.g. Java, Spring Boot, PostgreSQL, React',
  'Quản lý nhập tự do, ngăn cách bởi dấu phẩy. AI sẽ dùng để đối chiếu khi gợi ý.':
    'Free text, comma-separated. AI uses this for matching when making suggestions.',

  // --- Dự án ---
  'Bạn có chắc muốn xóa dự án này?': 'Are you sure you want to delete this project?',
  'Thêm dự án': 'Add project',
  'Tên dự án': 'Project name',
  'Ngày bắt đầu': 'Start date',
  'Ngày kết thúc': 'End date',
  'Chưa có dự án nào.': 'No projects yet.',
  'Chỉnh sửa dự án': 'Edit project',
  'Thêm dự án mới': 'Add new project',
  'Nhập tên dự án': 'Enter the project name',
  'Mô tả ngắn về dự án': 'Short project description',

  // --- Công việc ---
  'Bạn có chắc muốn xóa công việc này?': 'Are you sure you want to delete this task?',
  'Thêm công việc': 'Add task',
  'Tiêu đề': 'Title',
  'Hạn chót': 'Due date',
  'Phân công': 'Assignee',
  'Chưa có công việc nào.': 'No tasks yet.',
  'Chỉnh sửa công việc': 'Edit task',
  'Thêm công việc mới': 'Add new task',
  'Nhập tiêu đề công việc': 'Enter the task title',
  'Mô tả công việc': 'Task description',
  'Kỹ năng yêu cầu': 'Required skills',
  'VD: Java, Spring Boot, PostgreSQL': 'e.g. Java, Spring Boot, PostgreSQL',
  'Quản lý nhập tự do, AI sẽ đối chiếu với kỹ năng của nhân viên khi gợi ý.':
    'Free text; AI matches it against employee skills when making suggestions.',
  'Phân công cho': 'Assign to',
  '-- Chọn dự án --': '-- Select a project --',
  '-- Chọn nhân viên --': '-- Select an employee --',

  // --- Chấm công ---
  'Đã duyệt': 'Approved',
  'Chờ duyệt': 'Pending review',
  'Từ chối': 'Rejected',
  'Lọc theo nhân viên:': 'Filter by employee:',
  'Tất cả nhân viên': 'All employees',
  'Ngày': 'Date',
  'Giờ vào': 'Check-in',
  'Giờ ra': 'Check-out',
  'Khoảng cách': 'Distance',
  'Chưa có dữ liệu chấm công.': 'No attendance records yet.',
  'Duyệt': 'Approve',
  'Chấm công thất bại.': 'Failed to log attendance.',
  'Ghi nhận chấm công': 'Log attendance',

  // --- Công việc của tôi ---
  'Các nhiệm vụ được quản lý phân cho bạn. Bạn có thể cập nhật trạng thái công việc trực tiếp.':
    'Tasks assigned to you by your manager. You can update task status directly.',
  'Cập nhật': 'Update',
  'Bạn chưa được giao công việc nào.': 'You have not been assigned any tasks.',
  'Không tải được danh sách công việc của bạn.': 'Failed to load your task list.',
  'Cập nhật trạng thái thất bại.': 'Failed to update status.',

  // --- Chấm công của tôi ---
  'Hệ thống dùng GPS để xác minh vị trí. Nếu nằm ngoài vùng cho phép, bản ghi sẽ chuyển sang':
    'The system uses GPS to verify your location. If you are outside the allowed area, the record changes to',
  'trạng thái chờ quản lý duyệt': 'a status awaiting manager approval',
  'Vị trí hiện tại': 'Current location',
  'Đang lấy GPS…': 'Getting GPS…',
  'Gần nhất:': 'Nearest:',
  'Khoảng cách:': 'Distance:',
  'trong vùng': 'inside the area',
  'ngoài vùng cho phép': 'outside the allowed area',
  'Chưa có văn phòng nào được cấu hình. Liên hệ quản lý.':
    'No office has been configured. Contact your manager.',
  'Vào ca (Check-in)': 'Check in',
  'Tan ca (Check-out)': 'Check out',
  'Chưa có bản ghi chấm công nào.': 'No attendance records yet.',
  'Không tải được dữ liệu.': 'Failed to load data.',
  'Trình duyệt không hỗ trợ Geolocation API.': 'This browser does not support the Geolocation API.',
  'Không lấy được vị trí. Hãy cấp quyền GPS cho trình duyệt.':
    'Could not get your location. Please grant GPS permission to the browser.',
  'Chấm công vào': 'Check-in',
  'Chấm công ra': 'Check-out',
  '{label} thành công.': '{label} successful.',
  '{label} thành công nhưng vị trí ngoài vùng cho phép – đã chuyển sang chờ duyệt.':
    '{label} successful, but the location is outside the allowed area – moved to pending review.',
  '{label} thất bại.': '{label} failed.',

  // --- AI Gợi ý ---
  'AI Gợi ý nhân viên': 'AI employee suggestions',
  'Phân tích lịch sử và đề xuất người phù hợp nhất':
    'Analyzes history and recommends the best-fit people',
  'Tiêu đề công việc': 'Task title',
  'tùy chọn': 'optional',
  'Đang phân tích...': 'Analyzing...',
  'Phân tích bằng AI': 'Analyze with AI',
  'Không thể lấy gợi ý.': 'Could not get suggestions.',
  'Không tìm thấy nhân viên phù hợp': 'No suitable employee found',
  'Hãy thử mô tả task chi tiết hơn': 'Try describing the task in more detail',

  // --- Văn phòng & Geofence ---
  'Văn phòng & Geofence': 'Offices & Geofence',
  'Cấu hình điểm cho phép chấm công bằng GPS. Bán kính tính bằng mét.':
    'Configure points where GPS attendance is allowed. Radius is in meters.',
  'Thêm văn phòng': 'Add office',
  'Tên văn phòng': 'Name',
  'Địa chỉ': 'Address',
  'Toạ độ': 'Coordinates',
  'Bán kính': 'Radius',
  'Chưa có văn phòng nào.': 'No offices yet.',
  'Sửa văn phòng': 'Edit office',
  'Thêm văn phòng': 'Add office',
  'Tên *': 'Name *',
  'Bán kính (m)': 'Radius (m)',
  'Dùng vị trí hiện tại': 'Use current location',
  'Không tải được danh sách văn phòng.': 'Failed to load the office list.',
  'Xoá văn phòng "{name}"?': 'Delete office "{name}"?',
  'Xoá thất bại.': 'Delete failed.',
  'Bán kính: {radius}m': 'Radius: {radius}m',
  'Vị trí của bạn': 'Your location',

  // --- Người dùng & Phân quyền ---
  'Người dùng & Phân quyền': 'Users & Permissions',
  'Tài khoản mới đăng ký mặc định là Nhân viên. Quản trị viên đổi vai trò để cấp quyền quản lý.':
    'Newly registered accounts default to Employee. Administrators change roles to grant management rights.',
  'Tên đăng nhập': 'Username',
  'Vai trò hiện tại': 'Current role',
  'Phân quyền': 'Assign role',
  'Chưa có người dùng nào.': 'No users yet.',
  'Không thể đổi': 'Cannot change',
  'Đã đổi vai trò của "{name}" thành {role}.': 'Changed the role of "{name}" to {role}.',
  'Đổi vai trò thất bại.': 'Failed to change role.',

  // --- Modal / ErrorBoundary ---
  'Đã xảy ra lỗi': 'An error occurred',
  'Ứng dụng gặp sự cố không mong muốn. Vui lòng tải lại trang.':
    'The application ran into an unexpected problem. Please reload the page.',
  'Tải lại trang': 'Reload page',

  // --- Trang Landing & chế độ dùng thử ---
  'Quản lý công việc thông minh cho doanh nghiệp nhỏ':
    'Smart task management for small businesses',
  'Giao việc, theo dõi tiến độ, chấm công GPS và để AI gợi ý nhân viên phù hợp — tất cả trong một hệ thống.':
    'Assign work, track progress, GPS attendance, and let AI suggest the right people — all in one system.',
  'Dùng thử ngay': 'Try it now',
  'Chế độ dùng thử cho xem trước giao diện với dữ liệu mẫu — không cần tài khoản.':
    'Demo mode previews the interface with sample data — no account needed.',
  'Tính năng nổi bật': 'Key features',
  'Sẵn sàng trải nghiệm hệ thống?': 'Ready to explore the system?',
  'Đăng nhập để dùng đầy đủ, hoặc dùng thử ngay với dữ liệu mẫu.':
    'Sign in for full access, or try it now with sample data.',
  'Quản lý công việc & dự án': 'Task & project management',
  'Giao việc, theo dõi tiến độ, hạn chót và trạng thái.':
    'Assign work; track progress, deadlines and status.',
  'AI gợi ý nhân viên': 'AI employee suggestions',
  'Google Gemini phân tích và đề xuất người phù hợp nhất.':
    'Google Gemini analyzes and recommends the best-fit people.',
  'Chấm công GPS': 'GPS attendance',
  'Xác thực vị trí bằng bản đồ, hạn chế chấm công gian lận.':
    'Verify location on a map to reduce attendance fraud.',
  'Phân quyền & bảo mật': 'Roles & security',
  'Ba vai trò, xác thực JWT, dữ liệu được bảo vệ nhiều lớp.':
    'Three roles, JWT authentication, multi-layer data protection.',
  'Dashboard trực quan': 'Visual dashboard',
  'Biểu đồ thống kê công việc và nhân sự theo thời gian thực.':
    'Real-time charts of tasks and staff.',
  'Song ngữ Việt / Anh': 'Vietnamese / English',
  'Chuyển đổi ngôn ngữ giao diện tức thì, mọi lúc.':
    'Switch the interface language instantly, anytime.',
  'Chế độ dùng thử — bạn đang xem dữ liệu mẫu.':
    'Demo mode — you are viewing sample data.',
  'Đăng nhập để dùng đầy đủ': 'Sign in for full access',
  'Khách dùng thử': 'Demo guest',
  'Chế độ dùng thử': 'Demo mode',
  'Thoát dùng thử': 'Exit demo',
}

File: mobile/lib/i18n/translations.dart
/// Bảng dịch tiếng Anh. Khóa = chuỗi tiếng Việt gốc (giống quy ước bên frontend).
/// - locale 'vi' → trả về key (Vietnamese gốc)
/// - locale 'en' → tra cứu bảng dưới; không có thì fallback về key
/// - placeholder {name} → thay bằng tham số truyền vào hàm tr().
///
/// Khi cập nhật: thêm cả khóa Việt + bản dịch Anh; đừng tách rời.
const Map<String, String> en = {
  // --- Common navigation / actions ---
  'Bảng điều khiển': 'Dashboard',
  'Nhân viên': 'Employees',
  'Dự án': 'Projects',
  'Công việc': 'Tasks',
  'Chấm công': 'Attendance',
  'Văn phòng': 'Offices',
  'AI Gợi ý': 'AI Suggestions',
  'Công việc của tôi': 'My Tasks',
  'Chấm công của tôi': 'My Attendance',
  'Người dùng': 'Users',
  'Đăng xuất': 'Log out',
  'Đăng nhập': 'Log in',
  'Đăng ký': 'Sign up',
  'Đăng ký ngay': 'Sign up now',
  'Huỷ': 'Cancel',
  'Lưu': 'Save',
  'Xoá': 'Delete',
  'Sửa': 'Edit',
  'Tạo': 'Create',
  'Quay lại': 'Back',
  'Đang tải...': 'Loading...',
  'Chưa có dữ liệu': 'No data yet',
  'Có lỗi xảy ra': 'Something went wrong',

  // --- Auth ---
  'Quản lý Công việc': 'Task Manager',
  'Chào mừng trở lại': 'Welcome back',
  'Đăng nhập vào Task Manager': 'Log in to Task Manager',
  'Email': 'Email',
  'Mật khẩu': 'Password',
  'Vui lòng nhập email': 'Please enter your email',
  'Email không hợp lệ': 'Invalid email',
  'Vui lòng nhập mật khẩu': 'Please enter your password',
  'Đăng nhập thất bại': 'Login failed',
  'Chưa có tài khoản?': 'No account yet?',
  'Đã có tài khoản?': 'Already have an account?',

  // --- Register ---
  'Tạo tài khoản': 'Create account',
  'Tham gia hệ thống quản lý công việc': 'Join the task management platform',
  'Tên đăng nhập': 'Username',
  'Vui lòng nhập tên đăng nhập': 'Please enter a username',
  'Tối thiểu 3 ký tự': 'At least 3 characters',
  'Mật khẩu từ 8 đến 100 ký tự': 'Password must be 8–100 characters',
  'Phải có chữ, số và ký tự đặc biệt': 'Must contain a letter, a digit, and a special character',
  'Tối thiểu 8 ký tự, gồm chữ, số và ký tự đặc biệt (vd: !@#\$)':
      'At least 8 characters with letters, digits, and a special character (e.g. !@#\$)',
  'Đăng ký thành công! Vui lòng đăng nhập.': 'Registered successfully! Please log in.',
  'Đăng ký thất bại': 'Registration failed',

  // --- Dashboard / Stats ---
  'Xin chào, {name}': 'Hello, {name}',
  'Xin chào, {name}!': 'Hello, {name}!',
  'Tổng quan': 'Overview',
  'Tổng quan hệ thống': 'System overview',
  'Chấm công hôm nay': 'Attendance today',
  'Chấm công tháng này': 'Attendance this month',
  'Hoàn thành': 'Completed',
  'HOÀN THÀNH': 'COMPLETED',
  'Đang làm': 'In progress',
  'Đang thực hiện': 'In progress',
  'Chờ xử lý': 'Pending',
  'Quá hạn': 'Overdue',
  'Tỷ lệ hoàn thành: {pct}%': 'Completion rate: {pct}%',
  'Phân bố trạng thái công việc': 'Task status breakdown',
  '{total} công việc · {done} đã hoàn thành': '{total} tasks · {done} completed',
  'Đang tải dữ liệu...': 'Loading data...',
  'Đổi ngôn ngữ': 'Change language',
  'bạn': 'you',

  // --- Tasks screen ---
  'Tạo công việc mới': 'Create new task',
  'Tiêu đề *': 'Title *',
  'Bắt buộc': 'Required',
  'Mô tả': 'Description',
  'Hạn hoàn thành (YYYY-MM-DD)': 'Due date (YYYY-MM-DD)',
  'Đã tạo công việc "{title}"': 'Created task "{title}"',
  'Tạo công việc thất bại': 'Failed to create task',
  'Cập nhật trạng thái': 'Update status',
  'Đã cập nhật trạng thái': 'Status updated',
  'Cập nhật thất bại': 'Update failed',
  'Tất cả': 'All',
  'Đang tải danh sách công việc...': 'Loading tasks...',
  'Không có công việc nào': 'No tasks yet',
  'Đổi trạng thái': 'Change status',
  'Thử lại': 'Retry',

  // --- Employees screen ---
  'Tạo nhân viên mới': 'New employee',
  'Thêm nhân viên': 'Add employee',
  'Thêm': 'Add',
  'Họ *': 'Last name *',
  'Tên *': 'First name *',
  'Đã thêm nhân viên {name}': 'Added employee {name}',
  'Thêm thất bại': 'Failed to add',
  'Chưa có nhân viên nào': 'No employees yet',
  'Ngày vào: {date}': 'Hired: {date}',
  'Chức vụ': 'Position',
  'Phòng ban': 'Department',
  'Nhóm': 'Group',
  'Kỹ năng (phân cách bởi dấu phẩy)': 'Skills (comma-separated)',
  'Đã thêm nhân viên': 'Employee added',
  'Thêm nhân viên thất bại': 'Failed to add employee',
  'Đang tải danh sách nhân viên...': 'Loading employees...',
  'Không có nhân viên nào': 'No employees yet',
  'Xoá nhân viên này?': 'Delete this employee?',
  'Đã xoá nhân viên': 'Employee deleted',
  'Xoá thất bại': 'Delete failed',

  // --- Projects screen ---
  'Tạo dự án mới': 'New project',
  'Thêm dự án': 'Add project',
  'Tên dự án *': 'Project name *',
  'Bắt đầu: {date}': 'Start: {date}',
  'Kết thúc: {date}': 'End: {date}',
  'Chưa có dự án nào': 'No projects yet',
  'Đang tải danh sách dự án...': 'Loading projects...',
  'Đã tạo dự án {name}': 'Created project {name}',
  'Ngày bắt đầu (YYYY-MM-DD) *': 'Start date (YYYY-MM-DD) *',
  'Ngày kết thúc (YYYY-MM-DD)': 'End date (YYYY-MM-DD)',
  'Trạng thái': 'Status',
  'Đã tạo dự án': 'Project created',
  'Tạo dự án thất bại': 'Failed to create project',
  'Không có dự án nào': 'No projects yet',
  'Xoá dự án "{name}"?': 'Delete project "{name}"?',
  'Đã xoá dự án': 'Project deleted',
  'Đang triển khai': 'Ongoing',
  'Đã hoàn thành': 'Completed',
  'Đã huỷ': 'Cancelled',

  // --- Attendance screen ---
  'Vị trí GPS': 'GPS location',
  'Đang lấy GPS…': 'Getting GPS…',
  'Cập nhật': 'Refresh',
  'Lịch sử chấm công': 'Attendance history',
  'Chưa có bản ghi hôm nay': 'No records today yet',
  'Chưa có bản ghi': 'No records',
  'Nhân viên #{id}': 'Employee #{id}',
  '{office} – {dist}m (trong vùng)': '{office} – {dist}m (within range)',
  '{office} – {dist}m / {max}m (ngoài vùng, sẽ chờ duyệt)':
      '{office} – {dist}m / {max}m (out of range, will be queued for review)',
  'Chưa có văn phòng nào được cấu hình. Liên hệ quản lý.':
      'No office configured. Contact your manager.',
  'Check-in thành công': 'Check-in succeeded',
  'Check-out thành công': 'Check-out succeeded',
  'Check-in thành công – chờ quản lý duyệt': 'Check-in succeeded — pending review',
  'Check-out thành công – chờ quản lý duyệt': 'Check-out succeeded — pending review',
  'Lỗi: {error}': 'Error: {error}',
  'Vui lòng bật Dịch vụ Định vị (Location).': 'Please enable Location services.',
  'Bạn từ chối quyền truy cập vị trí.': 'Location permission was denied.',
  'Quyền vị trí bị tắt vĩnh viễn. Hãy bật trong cài đặt.':
      'Location permission permanently denied. Please enable it in settings.',
  'Tải xuống Excel': 'Download Excel',
  'Đang tải dữ liệu chấm công...': 'Loading attendance...',
  'Không có dữ liệu chấm công': 'No attendance data',
  'Duyệt chấm công': 'Review attendance',
  'Lý do từ chối (tuỳ chọn)': 'Rejection reason (optional)',
  'Duyệt': 'Approve',
  'Từ chối': 'Reject',
  'Đã duyệt': 'Approved',
  'Đã từ chối': 'Rejected',
  'Chờ duyệt': 'Pending review',
  'Khoảng cách: {m}m': 'Distance: {m}m',
  'Ngoài bán kính cho phép': 'Outside the allowed radius',
  'Trong phạm vi văn phòng': 'Within office radius',
  'Vào: {time}': 'In: {time}',
  'Ra: {time}': 'Out: {time}',

  // --- AI Suggestions screen ---
  'AI Gợi ý nhân viên': 'AI staff suggestions',
  'AI gợi ý nhân viên phù hợp': 'AI suggests the right people',
  'Phân tích kỹ năng, hiệu suất và workload': 'Analyzes skills, performance, and workload',
  'Tiêu đề công việc': 'Task title',
  'Vui lòng nhập tiêu đề': 'Please enter a title',
  'Mô tả (tùy chọn)': 'Description (optional)',
  'Đang phân tích...': 'Analyzing...',
  'Phân tích bằng AI': 'Analyze with AI',
  'AI đang phân tích...': 'AI is analyzing...',
  'AI gợi ý nhân sự': 'AI staff suggestions',
  'Mô tả công việc cần làm': 'Describe the task',
  'Vd: Cần dev React + Spring Boot làm trang quản lý đơn hàng':
      'E.g. Need a React + Spring Boot dev for an order management page',
  'Nhận gợi ý từ AI': 'Get AI suggestion',
  'Đang phân tích bằng AI...': 'Analyzing with AI...',
  'Chưa có gợi ý — nhập yêu cầu rồi nhấn nút bên trên':
      'No suggestions yet — enter a request and tap the button above',
  'Phù hợp {pct}%': '{pct}% match',
  'Lý do': 'Reason',
  'Kỹ năng phù hợp': 'Matching skills',
};

/// Thay placeholder {key} trong [template] bằng giá trị từ [params].
String _interpolate(String template, Map<String, Object?> params) {
  if (params.isEmpty) return template;
  var out = template;
  params.forEach((k, v) {
    out = out.replaceAll('{$k}', '$v');
  });
  return out;
}

/// Trả về bản dịch của [key] theo [locale]. Mặc định fallback về key gốc.
String translate(String locale, String key, [Map<String, Object?> params = const {}]) {
  if (locale == 'vi') return _interpolate(key, params);
  final translated = en[key];
  return _interpolate(translated ?? key, params);
}
